c/parallel/src/util/types.h

```diff
@@ -22,6 +22,33 @@ struct input_storage_t;
 struct output_storage_t;
 struct items_storage_t; // Used in merge_sort
 
+// On Windows, nvrtcGetTypeName calls UnDecorateSymbolName from Dbghelp.dll,
+// which, for certain input types, returns string representations that nvcc
+// balks on (e.g. `long long` becomes `__int64`).  This helper function looks
+// for these unsupported types and converts them to nvcc-compatible types.
+// The method signature is kept identical to `nvrtcGetTypeName` so that this
+// helper can be used as a drop-in replacement.
+template <typename T>
+nvrtcResult cccl_type_name_from_nvrtc(std::string* result)
+{
+  nvrtcResult res = nvrtcGetTypeName<T>(result);
+  if (res != NVRTC_SUCCESS)
+  {
+    return res;
+  }
```

Comment:
Suggestion: if you feel fancy today:
```suggestion
  if (const nvrtcResult res = nvrtcGetTypeName<T>(result); res != NVRTC_SUCCESS)
  {
    return res;
  }
```